pkg/flowkit/util/mixpanel.go

```diff
@@ -0,0 +1,170 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+const (
+	MIXPANEL_TRACK_URL              = "https://api.mixpanel.com/track"
+	MIXPANEL_QUERY_URL              = "https://mixpanel.com/api/2.0/engage?project_id=2763737"
+	MIXPANEL_PROFILE_URL            = "https://api.mixpanel.com/engage#profile-set"
+	MIXPANEL_PROJECT_TOKEN          = "7af4e6f44df2c77935477ba103b3c529"
+	MIXPANEL_SERVICE_ACCOUNT_SECRET = "Rmxvdy1jbGkuZTVkMTNjLm1wLXNlcnZpY2UtYWNjb3VudDp2TkZrVzhiWWNSY1ZuQmtMZFF4bXVzamdZa0dyc2FsMQ=="
+)
+
+type MixpanelClient struct {
+	token   string
+	baseUrl string
+}
+
+func TrackCommandUsage(command *cobra.Command) error {
+	mixpanelEvent := newEvent(command)
+	mixpanelEvent.setUpEvent(MIXPANEL_PROJECT_TOKEN, FLOW_CLI)
+	eventPayload, err := encodePayload(mixpanelEvent)
+	if err != nil {
+		return err
+	}
+	payload := strings.NewReader(eventPayload)
+	req, err := http.NewRequest("POST", MIXPANEL_TRACK_URL, payload)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("Accept", "text/plain")
+	req.Header.Add("Content-Type", "application/json")
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	_, err = ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		return err
+	}
+	if res.StatusCode >= 400 {
+		return fmt.Errorf("invalid response status code %d for tracking command usage", res.StatusCode)
+	}
+
+	return nil
+}
+
+func encodePayload(obj interface{}) (string, error) {
+	b, err := json.Marshal(obj)
+	if err != nil {
+		return "", err
+	}
+	formattedString := "[" + string(b) + "]"
+	return formattedString, nil
+}
+
+func SetUserTrackingSettings(enable bool) error {
+	mixpanelUser, err := getMixPanelUser()
+	if err != nil {
+		return err
+	}
+	mixpanelUser.configureUserTracking(enable)
+
+	userPayload, err := encodePayload(mixpanelUser)
+	if err != nil {
+		return err
+	}
+	payload := strings.NewReader(userPayload)
+	req, err := http.NewRequest("POST", MIXPANEL_PROFILE_URL, payload)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("Accept", "text/plain")
+	req.Header.Add("Content-Type", "application/json")
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	_, err = ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		return err
+	}
+	if res.StatusCode >= 400 {
+		return fmt.Errorf("invalid response status code %d for tracking command usage", res.StatusCode)
+	}
+
+	return nil
+}
+
+type MixPanelResponse struct {
+	Results []struct {
+		DistinctId string `json:"$distinct_id"`
+		Properties struct {
+			City     interface{} `json:"$city"`
```

Comment:
In Go1.18 you can use `any` instead of `interface{}`. It's a habit I still sometimes use interface but it's better going forward to just use any